lib/src/transactions.dart

```diff
@@ -80,10 +80,11 @@ class ArweaveTransactionsApi {
   }) async* {
     final uploader = await getUploader(transaction, forDataOnly: dataOnly);
 
+    if (!dryRun) {
+      uploader.batchUploadChunks();
+    }
     while (!uploader.isComplete) {
```

Comment:
This is still sort of an infinite loop. The way is should be written is:
```
if (!dryRun) {
  while (!uploader.isComplete) {
    uploader.batchUploadChunks();
    yield uploader;
  }
}
```

Reply:
Updated